lib/get_player_input.rb

```diff
@@ -0,0 +1,23 @@
+require_relative 'tic_tac_toe.rb'
+
+def get_player_input(input, player, board=Array.new(3){Array.new(3, "_")})
+    puts "Player #{player}, enter your placeholder in format num1, num2"
+    input = input.gets.chomp
```

Comment:
It might be useful to validate the user input. i.e row and col should only be digits from 0 to 2

I saw a validation about ensuring the cell doesn't get overridden if it already has a move. That's brilliant.

Reply:
Agreed on both of these points.  Great that you added some validation, so that the player is not able to pick a space on the board that is already occupied.  Returning `false` looks like a reasonable thing to do, for now.

And yes it would be helpful to separate the side-effect (a.k.a. "command") of updating the game board from the queries of getting user input and checking if the indicated space is already occupied.  A future `TicTacToeGame` class might do that part.